core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/OpenStackCredentialParam.java

```diff
@@ -6,7 +6,13 @@
 
     USER("user", true, String.class, Optional.<String>absent()),
     PASSWORD("password", true, String.class, Optional.<String>absent()),
-    TENANT_NAME("tenantName", true, String.class, Optional.<String>absent()),
+    TENANT_NAME("tenantName", false, String.class, Optional.<String>absent()),
+    USER_DOMAIN("userDomain", false, String.class, Optional.<String>absent()),
+    KEYSTONE_VERSION("keystoneVersion", true, String.class, Optional.<String>absent()),
+    KEYSTONE_AUTH_SCOPE("keystoneAuthScope", false, String.class, Optional.<String>absent()),
+    PROJECT_DOMAIN_NAME("projectDomainName", false, String.class, Optional.<String>absent()),
+    PROJECT_NAME("projectName", false, String.class, Optional.<String>absent()),
+    DOMAIN_NAME("domainName", false, String.class, Optional.<String>absent()),
     ENDPOINT("endpoint", true, String.class, Optional.<String>absent());
```

Comment:
Validation improvement would be good:

* In case of v2 the tenantName is mandatory
* In case of v3 some of the new fields are mandatory
* In case of v3 there are mandatory fields depends on the given scope